File: ai-rag-app/ai-rag-app/Store/AppState.swift
//
//  AppState.swift
//  ai-rag-app
//
//  Created by Graham Diehl on 8/26/26.
//

import Foundation

struct AppState: Equatable {
    var osha = OshaState()
}

struct OshaState: Equatable {
    var isLoading: Bool = false
    var response: OshaStandardResponse? = nil
    var errorMessage: String? = nil
    var route: [OshaRoute] = []
}

enum OshaRoute: Hashable {
    case response
}

File: ai-rag-android/app/src/main/java/net/apperdashery/demo/ai_rag_android/store/AppState.kt
package net.apperdashery.demo.ai_rag_android.store

import net.apperdashery.demo.ai_rag_android.model.OshaStandardResponse

data class AppState(
    val osha: OshaState = OshaState()
)

data class OshaState(
    val isLoading: Boolean = false,
    val response: OshaStandardResponse? = null,
    val errorMessage: String? = null,
    val route: OshaRoute? = null
)

enum class OshaRoute {
    Response
}
